#summary REPK manual
#labels Featured


=REPK Manual=

repk by Gabrielle Rocap and Eric Collins (rec3141.at.gmail.com)

==OVERVIEW==

REPK (*R*estriction *E*ndonuclease *P*ic*k*er) finds sets of 4 commercially available restriction endonucleases which together uniquely differentiate user-designated sequence groups from a user-supplied FASTA format sequence file of _any gene_, not just 16S rRNA.  The user-defined sequences groups allow the user to determine particular groups (e.g. taxonomic) of sequences to be differentiated by the program.  REPK is thus particularly useful in the case where a microbial community has been characterized by clone library analysis, as the actual sequences can be input into the program.  In this regard, shoving the entire RDP database through this program will a) certainly not work and b) be a waste of time -- to look at a very large community just use some common tetrameric restriction enzymes (e.g. see Engebretson and Moyer, 2003).  However, if you have a database of known sequences from an environment, this program could be useful to pick some enzymes that uniquely discriminate the different groups in your database.

==INPUTS==
===1. Multiple sequence alignment in FASTA format.===
All sequences should begin at the 5' end of your LABELED primer and end at the 3' end of your UNLABELED primer.  The sequences should have no gaps and have no whitespace in the taxa names. Unknown/bad sequence data should be replaced by X's or N's, and these areas should make up a small part of the sequences because they might have restriction sites that will be missed by this program.

  * Sequence groups are coded as characters of the taxa name and separated by a delimiter, and each sequence can have multiple delimiters.
  * You may find the [http://rdp.cme.msu.edu/classifier/classifier.jsp RDP classifier] useful in generating these subsets.

An example file might look like this:

{{{
>ARCH_EURY_PICRO_PICRO_Picrophilus_torridus_DSM9790
ACTCCGGTTGATCCTGCCGGCGGCCACTGCTATCAAGTTCCGACTAAGCC
>ARCH_EURY_TRMCC_PYROC_Pyrococcus_abyssi_GE5
ATTCCGGTTGATCCTGCCGGAGGCCACTGCTATGGGGGTCCGACTAAGCC
>ARCH_CREN_TMPTC_TMPRO_Thermoproteus_tenax
AAACCGGTTGATCCTGCCGGACCTGACCGCTATCGGGGTGGGGCTAAGCC
.
.
.
}}}


==OPTIONS==
===1. Restriction endonucleases and cut sites===
For now only the list from [http://rebase.neb.com/rebase/rebase.html REBASE] is available, in addition to a fake "no cut" enzyme.  Future possibilities will include user-defined and user-selected enzymes.

===2. Group Subset Delimiter===

The character used to separate groups in your FASTA sequence names, current choices are underscore (*_*) or pipe (*|*).  The example file above uses underscore group subset delimiters.  Default is underscore.

===3. Group Subset===
Selects group subset to differentiate.  Using the example file above, each delimiter will differentiate the sequences by:
  # Domain
  # Kingdom
  # Phylum
  # Order
  # Genera
  # Species
  # strain

For example, running the program with a 3 will search for enzymes that discriminate the different phyla in the dataset. If you want to to find enzymes that discriminate the different genera in the dataset, rerun using 5 as the group subset.  Default is 1.

===4. Cutoff===
This is the furthest apart (in basepairs) that two fragments can be in length and still be considered the same fragment.  This should be set considering such things as the amount of known variation in the sequences and the expected variability in fragment length determination during T-RFLP.  Default is 5.

===5. Stringency===
An individual enzyme must distinguish MORE than this percent of groups to be acceptable.  It is useful to increase this number (up to 100) if you get a lot of hits and the program takes a long time to execute.  Default is 50.

===6. Min/Max Fragment Lengths===
The shortest and longest fragment lengths acceptable.  You might change these depending on, e.g. the length of your PCR product, the mimimum or maximum fragment length your capillary electrophoresis machine can read accurately, the range of fragment lengths that survive your cleanup procedure, or the presence of primers and primer dimers. Any fragments shorter or longer than this won't be included in the analysis, but will be present in the 'fragfile' output.  Defaults are 75 and 900.

===7. Max Matches Returned===
The maximum number of results to return (helpful if you're lucky enough to have too many).  Will only return the highest ranking results.  Default is 100.

==OUTPUTS==

The program makes a new directory for every run and writes several files inside which can be accessed for up to 48 hours by bookmarking the unique URL given on the Results page.

===enzymes_type2.txt===
The REBASE enzyme list, if used in the analysis

===custom.txt===
The list of enzymes used in the analysis if selected from REBASE list or manuall inputted.

===success.txt===
This file lists the enzymes that singlehandedly distinguish all of the groups.

===no_cuts.txt===
This file lists the enzymes that don't cut/distinguish any of the groups.

===fragfile.txt===
This file shows the fragment lengths for each sequence and enzyme combination.

===enzmatrix.txt===
This is the file used by the program to calculate the best enzymes. Each column contains a pair of groups, and each row contains the enzyme name. A zero indicates that the enzyme does not distinguish those groups from each other. A one indicates that the enzyme does distinguish those groups from each other. The stringency setting specifies the fraction of ones that an enzyme must have to be acceptable.  This file can be useful in the case that there are no sets which uniquely differentiate your sequences.

===finalout.txt===
This is the final output which shows all of the successful enzyme groups.
  * "Enzyme Picker Key" is a key for the lower part.
    * Each numbered group of enzymes distinguishes the same groups of sequences (NOTE that they do not give the same fragment lengths, they only distinguish the same groups)
  * "Enzyme sets" shows the enzyme sets that uniquely differentiate your sequences
    * Score: the sum of the fractions of sequence groups discriminated by all of the enzymes (a perfect score equals 4).
  * Enzyme sets: each enzyme listed is only the first member of the set from "Enzyme Picker Key", so choosing any other member of the same group will yield the same discriminatory abilities.
    * Choose your enzyme based upon price, fragment lengths, whatever.


==REFERENCES==
Cole JR, Chai B, Farris RJ, Wang Q, Kulam SA, McGarrell DM, Garrity GM, Tiedje JM. The Ribosomal Database Project (RDP-II): sequences and tools for high-throughput rRNA analysis. Nucleic Acids Research 2005 Jan 1;33(Database Issue):D294-D296. doi: 10.1093/nar/gki038.

Engebretson JJ, Moyer CL (2003) Fidelity of select restriction endonucleases in determining microbial diversity by terminal-restriction fragment length polymorphism.  Applied and Environmental Microbiology 69:4823-4829.

Ludwig W, Strunk O, Westram R, Richter L, Meier H, Yadhu K, et al. (2004) ARB: a software environment for sequence data.  Nucleic Acids Research 32:1363â€“1371.

Marsh TL, Saxman P, Cole J, Tiedje J (2000) Terminal restriction fragment length polymorphism analysis program, a web-based research tool for microbial community analysis.

Ricke P, Kolb S, Braker G (2005) Application of a newly developed ARB software-integrated tool for in silico terminal restriction fragment length polymorphism analysis reveals the dominance of a novel pmoA cluster in a forest soil.  Applied and Environmental Microbiology 71:1671-1673.

Roberts RJ, Vincze T, Posfai J, Macelis D (2005) REBASE Restriction enzymes and DNA methyltransferases.  Nucleic Acids Research 33:D230-D232. 













